applications/sofa/gui/qt/RealGUI.cpp

```diff
@@ -1686,26 +1680,15 @@ void RealGUI::initViewer(BaseViewer* _viewer)
 
 //------------------------------------
 
-void RealGUI::parseOptions(const std::vector<std::string>& options)
+void RealGUI::parseOptions()
 {
-    for (unsigned int i=0; i<options.size(); ++i)
-    {
-        if (options[i] == "enableInteraction")
-            m_enableInteraction = true;
-        if (options[i] == "noViewers")
-            mCreateViewersOpt = false;
-        if (options[i].substr(0,4).compare("msaa") == 0)
-        {
-            std::string::size_type pos = options[i].find('=') + 1;
-            if(pos < options[i].npos)
-            {
-                std::string strNb = options[i].substr(pos);
-                m_viewerMSAANbSampling = atoi(strNb.c_str());
-                if(m_viewerMSAANbSampling < 2 || m_viewerMSAANbSampling > 32)
-                    m_viewerMSAANbSampling = 1;
-            }
-        }
-    }
+    po::variables_map vm = mArgumentParser->getVariableMap();
+    if(vm.find("enableInteraction") != vm.end())
+        m_enableInteraction = vm["enableInteraction"].as<bool>();
+    if(vm.find("noViewers") != vm.end()) /// TODO, missing args, what was the use ?
```

Comment:
Does someone know what is noviewer ? I did the syntax changes after the refactor, but "noviewer" is not a known option of runsofa/main.